pkg/ingester/client/ingester.pb.go

```diff
@@ -56,172 +60,17 @@ var MatchType_value = map[string]int32{
-	SkipLabelNameValidation bool                    `protobuf:"varint,1000,opt,name=skip_label_name_validation,json=skipLabelNameValidation,proto3" json:"skip_label_name_validation,omitempty"`
-}
-
-func (m *WriteRequest) Reset()      { *m = WriteRequest{} }
-func (*WriteRequest) ProtoMessage() {}
-func (*WriteRequest) Descriptor() ([]byte, []int) {
-	return fileDescriptor_893a47d0a749d749, []int{0}
-}
-func (m *WriteRequest) XXX_Unmarshal(b []byte) error {
-	return m.Unmarshal(b)
-}
-func (m *WriteRequest) XXX_Marshal(b []byte, deterministic bool) ([]byte, error) {
-	if deterministic {
-		return xxx_messageInfo_WriteRequest.Marshal(b, m, deterministic)
-	} else {
-		b = b[:cap(b)]
-		n, err := m.MarshalToSizedBuffer(b)
-		if err != nil {
-			return nil, err
-		}
-		return b[:n], nil
-	}
-}
-func (m *WriteRequest) XXX_Merge(src proto.Message) {
-	xxx_messageInfo_WriteRequest.Merge(m, src)
-}
-func (m *WriteRequest) XXX_Size() int {
-	return m.Size()
-}
-func (m *WriteRequest) XXX_DiscardUnknown() {
-	xxx_messageInfo_WriteRequest.DiscardUnknown(m)
-}
-
-var xxx_messageInfo_WriteRequest proto.InternalMessageInfo
-
-func (m *WriteRequest) GetSource() WriteRequest_SourceEnum {
-	if m != nil {
-		return m.Source
-	}
-	return API
-}
-
-func (m *WriteRequest) GetMetadata() []*MetricMetadata {
-	if m != nil {
-		return m.Metadata
-	}
-	return nil
-}
-
-func (m *WriteRequest) GetSkipLabelNameValidation() bool {
-	if m != nil {
-		return m.SkipLabelNameValidation
-	}
-	return false
-}
-
-type WriteResponse struct {
-}
-
-func (m *WriteResponse) Reset()      { *m = WriteResponse{} }
-func (*WriteResponse) ProtoMessage() {}
-func (*WriteResponse) Descriptor() ([]byte, []int) {
-	return fileDescriptor_893a47d0a749d749, []int{1}
-}
-func (m *WriteResponse) XXX_Unmarshal(b []byte) error {
-	return m.Unmarshal(b)
-}
-func (m *WriteResponse) XXX_Marshal(b []byte, deterministic bool) ([]byte, error) {
-	if deterministic {
-		return xxx_messageInfo_WriteResponse.Marshal(b, m, deterministic)
-	} else {
-		b = b[:cap(b)]
-		n, err := m.MarshalToSizedBuffer(b)
-		if err != nil {
-			return nil, err
-		}
-		return b[:n], nil
-	}
-}
-func (m *WriteResponse) XXX_Merge(src proto.Message) {
-	xxx_messageInfo_WriteResponse.Merge(m, src)
-}
-func (m *WriteResponse) XXX_Size() int {
-	return m.Size()
-}
-func (m *WriteResponse) XXX_DiscardUnknown() {
-	xxx_messageInfo_WriteResponse.DiscardUnknown(m)
+	return fileDescriptor_60f6df4f3586b478, []int{0}
 }
 
-var xxx_messageInfo_WriteResponse proto.InternalMessageInfo
-
 type ReadRequest struct {
 	Queries []*QueryRequest `protobuf:"bytes,1,rep,name=queries,proto3" json:"queries,omitempty"`
 }
 
 func (m *ReadRequest) Reset()      { *m = ReadRequest{} }
 func (*ReadRequest) ProtoMessage() {}
 func (*ReadRequest) Descriptor() ([]byte, []int) {
-	return fileDescriptor_893a47d0a749d749, []int{2}
+	return fileDescriptor_60f6df4f3586b478, []int{0}
```

Comment:
Do you know what's the 2nd returned parameter and if changing it is a breaking change?

Reply:
First returned value is a full description of "proto" file, which includes all types. It is itself a protobuf message, `FileDescriptorProto` and is described [here](https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/descriptor.proto). Second returned parameter is an index into `message_type` array.

Given that some messages were moved around, this proto file now has fewer messages, hence the change in returned index.